src/runtime_src/core/common/info_platform.cpp

```diff
@@ -352,6 +352,32 @@ add_clock_info(const xrt_core::device* device, ptree_type& pt)
   }
 }
 
+void
+add_tops_info(const xrt_core::device* device, ptree_type& pt)
+{
+    ptree_type pt_tops_array;
+
+    try {
+        auto resInfo = xrt_core::device_query<xq::xrt_resource_raw>(device);
```

Comment:
instead of try/catch, please use `device_query_default`

Reply:
but the return value is an array of multiple values. we don't have default values for them.